src/main/java/core/basesyntax/dao/FileReaderImpl.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.dao;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FileReaderImpl implements FileReader {
+
+    @Override
+    public List<String> readFile(String fileName) {
+        List<String> lines = new ArrayList<>();
+        try (BufferedReader br = new BufferedReader(new java.io.FileReader(fileName))) {
```

Comment:
Make sure to use try-with-resources with AutoCloseable resources. Here, you are using it correctly, which ensures that the BufferedReader is closed even if an exception occurs.